server/xpub-controller/manuscript.js

```diff
@@ -46,37 +36,21 @@ class Manuscript {
     // Predict upload time - The analysis was done on #839
     const predictedTime = 5 + 4.67e-6 * fileSize
     const startedTime = Date.now()
-
-    const handle = setInterval(() => {
-      const elapsed = Date.now() - startedTime
-      let progress = parseInt((100 * elapsed) / 1000 / predictedTime, 10)
-      // don't let the prediction complete the upload
-      if (progress > 99) progress = 99
-      pubsub.publish(`${ON_UPLOAD_PROGRESS}.${manuscriptId}`, {
-        manuscriptUploadProgress: progress,
-      })
-    }, 200)
+    const progress = FilesHelper.startFileProgress(
+      pubsub,
+      ON_UPLOAD_PROGRESS,
+      fileSize,
+      startedTime,
+      predictedTime,
+      manuscriptId,
+    )
 
```

Comment:
Could everything from this line (47) to where the manuscript has been successfully uploaded on line 99 be factored out into a new Manuscript Helper class which sits along side the File helper class. This class would then have a single function called uploadManuscript. This would split the porocess of Tracking the progress of an upload apart from the actual uploading of the file. This would then make it easier for us to wrap the whole uploading process in a try catch to clear the progress interval if something goes wrong. 

To elaborate further some sudo-code:
```
upload() {
  FileHelper.checkFileSize();
  FileHelper.startFileProgress();
  try {
    ManuscriptHelper.uploadManuscriptFile();
  } catch (error) {
    FileHelper.endFileProgress()
    throw error
  }
  return manuscript
}
```

Reply:
agreed

Reply:
Maybe leave out the try catch added for #1521 to take care of so this remains just a refactor task

Reply:
makes sense